eventstore/query_json.go

```diff
@@ -0,0 +1,232 @@
+package eventstore
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+
+	dsquery "github.com/ipfs/go-datastore/query"
+)
+
+type JSONQuery struct {
+	Ands []JSONCriterion
+	Ors  []JSONQuery
+	Sort JSONSort
+}
+type JSONCriterion struct {
+	FieldPath string
+	Operation JSONOperation
+	Value     JSONValue
+}
+
+type JSONValue struct {
+	String *string
+	Bool   *bool
+	Float  *float64
+}
+
+type JSONSort struct {
+	FieldPath string
+	Desc      bool
+}
+
+type JSONOperation int
+
+const (
+	Eq = JSONOperation(eq)
+	Ne = JSONOperation(ne)
+	Gt = JSONOperation(gt)
+	Lt = JSONOperation(lt)
+	Ge = JSONOperation(ge)
+	Le = JSONOperation(le)
+)
```

Comment:
This represents the data-structure used to make querys.
The operators enum values map directly to internal ones.